### PR TITLE
[NPM] Remove tracer internal stats from module stats

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?

Removes internal stats from getting reported as module stats. This also removes them from expvar reporting and from the `/debug/stats` endpoint. We are already reporting these in the `/telemetry` endpoint and they get picked up by the openmetrics/prometheus check in the agent.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/tracer/tracer_linux_test.go [hunk 1]
```diff
@@ -1904,7 +1904,7 @@ func (s *TracerSuite) TestGetHelpersTelemetry() {
 		syscall.Syscall(syscall.SYS_MUNMAP, uintptr(addr), uintptr(syscall.Getpagesize()), 0)
 	})
 
-	stats, err := tr.GetStats()
+	stats, err := tr.getStats(bpfMapStats)
```
Comment (usamasaqib) [inline pkg/network/tracer/tracer_linux_test.go:1907]:
```suggestion
	stats, err := tr.getStats(bpfHelperStats)
```

File: pkg/network/tracer/tracer_classification_linux_test.go [hunk 2]
```diff
@@ -32,35 +31,6 @@ func testProtocolClassificationInner(t *testing.T, params protocolClassification
...
-				}
-				if kStatsPost == nil {
-					return
-				}
-				for k, pre := range kStatsPre {
-					if !strings.HasSuffix(k, "_misses") {
-						continue
-					}
-					if post, ok := kStatsPost[k]; ok && pre != post {
-						t.Logf("kprobe stat %s differs pre=%v post=%v", k, pre, post)
```
Comment (brycekahle) [inline pkg/network/tracer/tracer_classification_linux_test.go:57]: It should be possible to still do this in this test, by directly calling `pkg/ebpf.GetProbeStats()`

Comment (hmahmood) [reply pkg/network/tracer/tracer_classification_linux_test.go:57]: Yes, but this was added by me for some debugging back when we were trying to figure out failing usm tests. Its not needed now.

**Review by brycekahle — APPROVED** [review]

**Review by akarpz — APPROVED** [review]

**Review by guyarb — APPROVED** [review]

**Review by richl-dd — APPROVED** [review]

**Pull request merged**